docs/operators/csvformatter.md

```diff
@@ -0,0 +1,133 @@
+CsvFormatter
+============
+
+## Operator Objective
+This operator receives a POJO ([Plain Old Java Object](https://en.wikipedia.org/wiki/Plain_Old_Java_Object)) as an incoming tuple, converts the data in 
+the incoming POJO to a custom delimited string and emits the delimited string.
+
+CsvFormatter supports schema definition as a JSON string. 
+
+CsvFormatter does not hold any state and is **idempotent**, **fault-tolerant** and **statically/dynamically partitionable**.
+
+## Operator Information
+1. Operator location: ***malhar-contrib***
+2. Available since: ***3.2.0***
+3. Operator state: ***Evolving***
+3. Java Packages:
+    * Operator: ***[com.datatorrent.contrib.formatter.CsvFormatter](https://www.datatorrent.com/docs/apidocs/com/datatorrent/contrib/formatter/CsvFormatter.html)***
+    
+## Properties, Attributes and Ports
+### <a name="props"></a>Properties of POJOEnricher
+| **Property** | **Description** | **Type** | **Mandatory** | **Default Value** |
+| -------- | ----------- | ---- | ------------------ | ------------- |
+| *schema* | Contents of the schema.Schema is specified in a json format. | String | Yes | N/A |
+
+
+### Platform Attributes that influences operator behavior
+| **Attribute** | **Description** | **Type** | **Mandatory** |
+| -------- | ----------- | ---- | ------------------ |
+| *in.TUPLE_CLASS* | TUPLE_CLASS attribute on input port which tells operator the class of POJO which will be incoming | Class or FQCN| Yes |
+
+
+### Ports
+| **Port** | **Description** | **Type** | **Mandatory** |
+| -------- | ----------- | ---- | ------------------ |
+| *in* | Tuples which need to be formatted are received on this port | Object (POJO) | Yes |
+| *out* | Tuples that are formatted are emitted from this port | String | Yes |
```

Comment:
The out port is described as mandatory but is not annotated that way in the base class .